src/features/offer/components/OfferContent/OfferContent.web.tsx

```diff
@@ -0,0 +1,100 @@
+import { OfferCTAButton } from 'features/offer/components/OfferCTAButton/OfferCTAButton'
+import { OfferPreviewModal } from 'features/offer/components/OfferPreviewModal/OfferPreviewModal'
+import { getOfferImageUrls } from 'features/offer/helpers/getOfferImageUrls/getOfferImageUrls'
+import { useOfferBatchTracking } from 'features/offer/helpers/useOfferBatchTracking/useOfferBatchTracking'
+import { OfferContentProps } from 'features/offer/types'
+import { useFeatureFlag } from 'libs/firebase/firestore/featureFlags/useFeatureFlag'
+import { RemoteStoreFeatureFlags } from 'libs/firebase/firestore/types'
+import { useGetHeaderHeight } from 'ui/components/headers/PageHeaderWithoutPlaceholder'
+import { useModal } from 'ui/components/modals/useModal'
+import { getSpacing } from 'ui/theme'
+
+export const OfferContent: FunctionComponent<OfferContentProps> = ({
+  offer,
+  searchGroupList,
+  subcategory,
+}) => {
+  const showOfferPreview = useFeatureFlag(RemoteStoreFeatureFlags.WIP_OFFER_PREVIEW)
+  const { visible, showModal, hideModal } = useModal(false)
+  const { isDesktopViewport } = useTheme()
+  const headerHeight = useGetHeaderHeight()
+  const [carouselDefaultIndex, setCarouselDefaultIndex] = useState(0)
+
+  const offerImages = useMemo(
+    () => (offer.images ? getOfferImageUrls(offer.images) : []),
+    [offer.images]
+  )
+
+  const { trackEventHasSeenOfferOnce } = useOfferBatchTracking({
+    offerNativeCategory: subcategory.nativeCategoryId,
+  })
+
+  const handlePress = (defaultIndex = 0) => {
+    if (showOfferPreview) {
+      setCarouselDefaultIndex(defaultIndex)
+      showModal()
+    }
+  }
+
+  const footer = isDesktopViewport ? null : (
+    <OfferCTAButton
+      offer={offer}
+      subcategory={subcategory}
+      trackEventHasSeenOfferOnce={trackEventHasSeenOfferOnce}
+    />
+  )
+
+  return (
+    <React.Fragment>
+      <OfferPreviewModal
+        hideModal={hideModal}
+        isVisible={visible}
+        offerImages={offerImages}
+        defaultIndex={carouselDefaultIndex}
+      />
+      <StyledOfferContentBase
+        showOfferPreview={showOfferPreview}
+        offer={offer}
+        searchGroupList={searchGroupList}
+        subcategory={subcategory}
+        isDesktopViewport={isDesktopViewport}
+        onOfferPreviewPress={handlePress}
+        footer={footer}>
+        {(body) => (
+          <BodyWrapper
+            isDesktopViewport={isDesktopViewport}
+            headerHeight={headerHeight}
+            testID={isDesktopViewport ? 'offer-body-desktop' : 'offer-body-mobile'}>
+            {body}
+          </BodyWrapper>
+        )}
+      </StyledOfferContentBase>
+    </React.Fragment>
+  )
+}
+
+const StyledOfferContentBase = styled(OfferContentBase).attrs<{
+  isDesktopViewport?: boolean
+}>(({ isDesktopViewport }) => ({
+  contentContainerStyle: { paddingBottom: isDesktopViewport ? 0 : getSpacing(22) },
+}))<{ isDesktopViewport?: boolean }>``
```

Comment:
il me semble que le theme est passé en argument des attributs et de la fonction du style de styled-components

```suggestion
const StyledOfferContentBase = styled(OfferContentBase).attrs(({ theme }) => ({
  contentContainerStyle: { paddingBottom: theme.isDesktopViewport ? 0 : getSpacing(22) },
}))``
```